alica_engine/include/engine/AlicaContext.h

```diff
@@ -125,7 +126,7 @@ class AlicaContext
      *
      * @note This is the main alica api class
      */
-    AlicaContext(const std::string& roleSetName, const std::string& masterPlanName, bool stepEngine);
+    AlicaContext(const std::string& roleSetName, const std::string& masterPlanName, bool stepEngine, AgentIDConstPtr agentID = nullptr);
```

Comment:
Would be better to take the agent Id as an integer parameter instead of agentId class, considering that the agent id in the alica config file is also specified to be an integer. It would also be easier for the user to use the API.

Reply:
In the case of a long or integer id, the default value cannot be nullptr. As a consequence, we would need to forbid some specific id value (e.g. -1) to be chosen by the application, in order to signal to the Alica engine that no id was set, if the value is -1. 

Therefore, I think that a null-able type is safer. Although the application needs to create an AgentIDConstPtr, which isn't that easy.

Reply:
Well, uint64 could be a valid type and 0 meaning nothing was given. But didn't we thought of taking AgentId object instead which internally could be empty? So application can pass something like AlicaContext(......., AgentId(3345) ) .
We might have to overload AgentId constructor to do this. And as far as I remember the bool operator is already overloaded.

Reply:
Added changes as discussed and documented in https://docs.google.com/presentation/d/13TxKVMJgYloT6HifuS5WFgAE8iFoJ88KBbJBlU8HOrQ/edit#slide=id.g72a0f35744_0_6